src/parser.rs

```diff
@@ -72,16 +75,20 @@ named!(fname<String>, alt!(quoted | bare));
  * Header lines
  */
 
-named!(header_line_content<String>,
+named!(header_line_content<(String, Option<DateTime<FixedOffset>>)>,
     chain!(
         filename: fname ~
-        take_until!("\n") ,
+        opt!(space) ~
+        after: map_res!(
+            take_until!("\n"),
+            std::str::from_utf8
+        ) ,
 
-        || filename
+        || (filename, DateTime::parse_from_str(after, "%F %T%.f %z").or_else(|_| DateTime::parse_from_str(after, "%F %T %z")).ok())
```

Comment:
the `.ok()` at the end here to just get a nice `Option` out for all input is slick :)